### ISSUE TITLE
Gentoo - stopped working on Vulkan titles after upgrade to Vulkan 1.3.250

### ISSUE THREAD

**Issue by Zi0P4tch0** [flightlessmango/MangoHud#1084]
**Describe the bug**

- Did a system wide upgrade on Gentoo Linux
- Vulkan tools, headers and loader got upgraded from 1.3.246 to 1.3.250
- MangoHud runs fine on `glxlgears` (logs message are visible in the console)
- MangoHud stopped working on any other Vulkan-based application - doesn't print messages to the console

**List relevant hardware/software information**

**MangoHud v0.6.9-r1**

```
System:
  Host: BrightFalls Kernel: 6.4.7-gentoo-x86_64 arch: x86_64 bits: 64
    Desktop: KDE Plasma v: 5.27.6 Distro: Gentoo Base System release 2.13
CPU:
  Info: 8-core model: AMD Ryzen 7 5800X3D bits: 64 type: MT MCP cache:
    L2: 4 MiB
  Speed (MHz): avg: 2215 min/max: 2200/4549 cores: 1: 2200 2: 2196 3: 2506
    4: 2170 5: 2200 6: 2198 7: 2199 8: 2200 9: 2194 10: 2200 11: 2194 12: 2198
    13: 2200 14: 2199 15: 2198 16: 2197
Graphics:
  Device-1: AMD Navi 21 [Radeon RX 6800/6800 XT / 6900 XT] driver: amdgpu
    v: kernel
  Display: wayland server: X.org v: 1.21.1.8 with: Xwayland v: 23.1.2
    compositor: kwin_wayland driver: X: loaded: amdgpu dri: radeonsi gpu: amdgpu
    resolution: 1: 2560x1440~144Hz 2: 1440x2560~60Hz
  API: OpenGL v: 4.6 Mesa 23.1.4 renderer: AMD Radeon RX 6800 XT (navi21
    LLVM 16.0.6 DRM 3.52 6.4.7-gentoo-x86_64)

```

**To Reproduce**
- Upgrade Vulkan tools, headers and loader from 1.3.246 to 1.3.250 on Gentoo Linux

**Expected behavior**
I've expected the MangoHud overlay to keep working on Vulkan games.

**Additional context**
A temporary fix was to downgrade the packages with the following mask:

```
>media-libs/vulkan-loader-1.3.246
>dev-util/vulkan-headers-1.3.246
>dev-util/vulkan-tools-1.3.246
```

Downgrading packages makes MangoHud working again for Vulkan titles.

Thanks!

**Comment by flightlessmango** [comment]
On my end I had to reconfigure and compile mangohud with the new libs and it worked fine again

**Comment by evelikov** [comment]
Sounds like a bug in mangohud. Either the python generator scripts are picking some odd version specific constant/macro, or the code itself is.

**Comment by WPettersson** [comment]
I am having the same issue on Gentoo. Upgraded to the 1.3.250 versions of vulkan-{loader,headers,tools} and while MangoHud would still print logging info relating to config file parsing to stdout, it wouldn't show an overlay and (according to in-game FPS meters) it wouldn't limit the FPS. I tested this in Sea of Thieves and Forza Horizon 5. I tried recompiling MangoHud-0.6.9.1 but it didn't change anything. However, dropped the vulkan versions back to 1.3.246 makes MangoHud work as expected again even without recompiling.

**Comment by flightlessmango** [comment]
Can you post a log with `VK_LOADER_DEBUG=all` and version 1.3.250 running vkcube?

**Comment by WPettersson** [comment]
Wasn't sure which you wanted, so
* https://strudel-hound.com/cube-1.3.246.log - vkcube on 1.3.246 without mangohud
* https://strudel-hound.com/mangohud-cube-1.3.246.log - vkcube on 1.3.246 with mangohud
* https://strudel-hound.com/cube-1.3.250.log - vkcube on 1.3.250 without mangohud
* https://strudel-hound.com/mangohud-cube-1.3.250.log - vkcube on 1.3.250 with mangohud

At a quick glance, I'm guessing `INFO:             /usr/lib/libMangoHud.so: wrong ELF class: ELFCLASS32` is a problem. I checked, gentoo installs both `/usr/bin/i686-pc-linux-gnu-vkcube` and `/usr/bin/x86_64-pc-linux-gnu-vkcubepp` on version 1.3.250. `mangohud /usr/bin/i686-pc-linux-gnu-vkcube` works fine, but `mangohud /usr/bin/x86_64-pc-linux-gnu-vkcubepp` doesn't show the hud / shows the wrong ELF class error. I can confirm that `/usr/lib/libMangoHud.so` is a 32-bit ELF library, while `/usr/lib64/libMangoHud.so` is a 64-bit ELF library.

**Comment by flightlessmango** [comment]
Yeah even tho he finds both 64 and 32bit manifests he's only trying the 32bit one.
What's the content of `MangoHud.x86_64.json` and `MangoHud.x86.json`?

**Comment by WPettersson** [comment]
```
% cat MangoHud.x86_64.json
{
    "file_format_version" : "1.0.0",
    "layer" : {
      "name": "VK_LAYER_MANGOHUD_overlay",
      "type": "GLOBAL",
      "api_version": "1.3.0",
      "library_path": "/usr/lib64/libMangoHud.so",
      "implementation_version": "1",
      "description": "Vulkan Hud Overlay",
      "functions": {
         "vkGetInstanceProcAddr": "overlay_GetInstanceProcAddr",
         "vkGetDeviceProcAddr": "overlay_GetDeviceProcAddr"
      },
      "enable_environment": {
        "MANGOHUD": "1"
      },
      "disable_environment": {
        "DISABLE_MANGOHUD": "1"
      }
    }
}
```

```
% cat MangoHud.x86.json
{
    "file_format_version" : "1.0.0",
    "layer" : {
      "name": "VK_LAYER_MANGOHUD_overlay",
      "type": "GLOBAL",
      "api_version": "1.3.0",
      "library_path": "/usr/lib/libMangoHud.so",
      "implementation_version": "1",
      "description": "Vulkan Hud Overlay",
      "functions": {
         "vkGetInstanceProcAddr": "overlay_GetInstanceProcAddr",
         "vkGetDeviceProcAddr": "overlay_GetDeviceProcAddr"
      },
      "enable_environment": {
        "MANGOHUD": "1"
      },
      "disable_environment": {
        "DISABLE_MANGOHUD": "1"
      }
    }
}
```

**Comment by flightlessmango** [comment]
`"name": "VK_LAYER_MANGOHUD_overlay"`
they both have the same name so the vulkan loader thinks it's the same one and will only use one. But since this commit 5754ca1 it should have separate names

**Comment by WPettersson** [comment]
Nice find. That commit isn't in the gentoo package, as it's not in the latest release (0.6.9-1), but I took the commit as a patch and applied it, and mangohud works fine now :)

**Comment by cronyakatsuki** [comment]
Okay then I see, but for some reason the overlay doesn't show for games like Deus Ex: Human revolution Director's Cut it doesn't show when only build with abi_x86_32, will try to also do abi_x86_64.

Could you provide the patch you used to fix this issue by any chance?

**Comment by WPettersson** [comment]
> Okay then I see, but for some reason the overlay doesn't show for games like Deus Ex: Human revolution Director's Cut when only build with abi_x86_32, will try to also do abi_x86_64.

Did DE:HR work before 1.3.250? 

> Could you provide the patch you used to fix the issue of multiple overlays by any chance?

It's exactly commit 5754ca1 from this repo - put a .patch at the end of that link and it'll give you the patch file

**Comment by cronyakatsuki** [comment]
Yes now version 0.6.9.1 works with the patch applied so it is an issue related to vulkan update to 1.3.250 and fact you need abi_x86_64 and abi_x86_32 to make sure all games work.

**Comment by abelsromero** [comment]
Similar issue in Manjaro. `mangohud glxgears` works fine, but `mangohud vkcube` does nothing.
I installed from source, commit efd1c07391c2aefc4af52d3b44e271ce37f782ef
